core/trino-main/src/main/java/io/trino/operator/output/PagePartitionerPool.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.output;
+
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class PagePartitionerPool
+{
+    private final Supplier<PagePartitioner> pagePartitionerSupplier;
+    @GuardedBy("this")
+    private final Queue<PagePartitioner> free = new LinkedList<>();
```

Comment:
Good point. Currently, the memory is accounted for more or less if there is an operator using the PagePartitioner but not otherwise. I will change that to pipeline-scoped memory context.

Reply:
Done. I used task-scoped memory context.